xen-hvm-stubdom-legacy.spec.in

```diff
@@ -0,0 +1,291 @@
+
+# Qubes specific patches
+Patch1001: patch-stubdom-vbd-non-dom0-backend.patch
+Patch1002: patch-xen-no-downloads.patch
+Patch1003: patch-xen-hotplug-external-store.patch
+Patch1004: patch-xen-stubdom-qubes-gui.patch
+Patch1005: patch-stubdom-lwip-fix-for-dhcp.patch
+Patch1006: patch-xen-libxl-qubes-minimal-stubdom.patch
+Patch1007: patch-xen-disable-dom0-qemu.patch
+Patch1009: patch-xenconsoled-enable-logging.patch
+#TODO Patch1110: patch-stubdom-linux-libxl-suspend.patch
+Patch1011: patch-xen-hotplug-qubesdb-update.patch
+Patch1012: patch-tools-hotplug-drop-perl-usage-in-locking-mechanism.patch
+Patch1013: patch-stubdom-linux-libxl-use-EHCI-for-providing-tablet-USB-device.patch
+Patch1014: patch-allow-kernelopts-stubdom.patch
+Patch1015: patch-libxl-readonly-disk-scsi.patch
+Patch1016: patch-tools-xenconsole-replace-ESC-char-on-xenconsole-outp.patch
+
+Patch1020: patch-stubdom-linux-config-qubes-gui.patch
+Patch1021: patch-stubdom-linux-libxl-do-not-force-qdisk-backend-for-cdrom.patch
+Patch1022: patch-xen-acpi-slic-support.patch
+
+Patch1100: patch-fix-errors-on-ambiguous-python-shebang.patch
+
+BuildRequires: xen-devel = %{epoch}:%{version}-%{release}
+BuildRequires: bison
+BuildRequires: flex
+BuildRequires: autoconf
+BuildRequires: automake
+BuildRequires: gettext
+BuildRequires: glibc-devel
+BuildRequires: gnutls-devel
+BuildRequires: openssl-devel
+BuildRequires: python2-devel
+BuildRequires: ncurses-devel gtk2-devel libaio-devel nettle-devel
+BuildRequires: pciutils-devel
+BuildRequires: libuuid-devel
+BuildRequires: acpica-tools
+BuildRequires: bzip2-devel xz-devel
+BuildRequires: yajl-devel
+BuildRequires: cmake
+BuildRequires: edk2-ovmf
+BuildRequires: dev86
+
+# so that x86_64 builds pick up glibc32 correctly
+BuildRequires: /usr/include/gnu/stubs-32.h
+
+%description
+Legacy stubdom files for Xen
+
+
+%prep
+%autosetup -p1 -n xen-%{upstream_version}
+
+# stubdom sources
+cp -v %{SOURCE10} %{SOURCE11} %{SOURCE12} %{SOURCE13} %{SOURCE14} stubdom
+cp -v %{SOURCE15} %{SOURCE16} %{SOURCE18} stubdom
+
+# qubes specific parts of stubdom
+tar xf %{SOURCE33}
+mkdir tools/qubes-gui/
+cp -a gui-agent-xen-hvm-stubdom/* tools/qubes-gui/
+
+tar xf %{SOURCE36}
+cp -a gui-common/include/qubes-gui*.h tools/qubes-gui/include/
+make -C tools/qubes-gui clean
+
+tar xf %{SOURCE34}
+cp -a core-vchan-xen/vchan tools/
+make -C tools/vchan -f Makefile.stubdom clean
+sed -e 's/ioemu-qemu-xen/qemu-xen-traditional/g' tools/qubes-gui/gui-agent-qemu/qemu-glue.patch | patch -p1
+
+tar xf %{SOURCE35}
+cp -a stubdom-dhcp/* tools/qemu-xen-traditional/
+patch -d tools/qemu-xen-traditional -p4 < stubdom-dhcp/lwip-dhcp-qemu-glue.patch
+
+
+%build
+
+%define efi_flags EFI_VENDOR=qubes
+mkdir -p dist/install/boot/efi/efi/qubes
+
+export XEN_VENDORVERSION="-%{release}"
+export EXTRA_CFLAGS_XEN_TOOLS="$RPM_OPT_FLAGS"
+export EXTRA_CFLAGS_QEMU_TRADITIONAL="$RPM_OPT_FLAGS"
+export EXTRA_CFLAGS_QEMU_XEN="$RPM_OPT_FLAGS"
+export PYTHON="/usr/bin/python2"
```

Comment:
Have you tried with python3?

Reply:
Yes with new newer commits on 'upstream' branch.